--- CMakeLists.txt	2025-03-20 02:27:55.192578503 -0500
+++ CMakeLists.txt	2025-03-20 02:28:44.520566682 -0500
@@ -941,7 +941,7 @@
 if("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
   list(APPEND ZIG_BUILD_ARGS -Doptimize=Debug)
 elseif("${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo")
-  list(APPEND ZIG_BUILD_ARGS -Doptimize=ReleaseFast)
+  list(APPEND ZIG_BUILD_ARGS -Doptimize=ReleaseSafe)
 else()
   list(APPEND ZIG_BUILD_ARGS -Doptimize=ReleaseFast -Dstrip)
 endif()
